src/main/groovy/jenkins/automation/utils/ScmUtils.groovy

```diff
@@ -57,4 +60,31 @@ class ScmUtils {
             }
         }
     }
+
+/***
+ *
+ * Utility method to do git shallow-clones easily.
+ * @param context A reference to the job object being modified
+ * @param url A string containing a git repo URL
+ * @param branch Optional branch to check-out; defaults to master
+ *
+ */
+    static void shallowGit(context, url, branch = "master") {
+        context.with {
+            git {
+                remote {
+                    delegate.url(url)
```

Comment:
Ah, that'd be great if it can be removed. I'll give it a shot. (I think I'm still pretty shaky when it comes to scope/binding/closures/etc in Groovy -- and how they all come together in job-dsl.)